src/nodes/NodeConnectionManager.ts

```diff
@@ -1745,19 +1784,39 @@ class NodeConnectionManager {
    * Returns a promise that resolves once the connection has authenticated,
    * otherwise it rejects with the authentication failure
    * @param nodeId
+   * @param ctx
    */
-  public async isAuthenticatedP(nodeId: NodeId): Promise<void> {
+  public async isAuthenticatedP(
+    nodeId: NodeId,
+    ctx?: Partial<ContextTimedInput>,
+  ): Promise<void>;
+  @timedCancellable(true)
+  public async isAuthenticatedP(
+    nodeId: NodeId,
+    @context ctx: ContextTimed,
+  ): Promise<void> {
     const targetNodeIdString = nodeId.toString() as NodeIdString;
     const connectionsEntry = this.connections.get(targetNodeIdString);
     if (connectionsEntry == null) {
       throw new nodesErrors.ErrorNodeConnectionManagerConnectionNotFound();
     }
+    const { p: abortP, rejectP: triggerAbort } = utils.promise<void>();
+    const abortHandler = () => {
+      triggerAbort(ctx.signal.reason);
+    };
+    if (ctx.signal.aborted) {
+      triggerAbort(ctx.signal.reason);
+    } else {
```

Comment:
Check if its aborted and then just throw immediately. Then you can create the promise, abort handler and hook it all up. It should come out cleaner that way.